src/main/scala/uk/gov/hmrc/play/microservice/bootstrap/JsonErrorHandling.scala

```diff
@@ -32,18 +33,37 @@ trait JsonErrorHandling extends GlobalSettings {
   implicit val erFormats = Json.format[ErrorResponse]
 
   override def onError(request: RequestHeader, ex: Throwable) = {
-    Future.successful {
-      val (code, message) = ex match {
-        case e: HttpException => (e.responseCode, e.getMessage)
+    logError(request, ex)
+    Future.successful(resolveError(request, ex))
+  }
 
-        case e: Upstream4xxResponse => (e.reportAs, e.getMessage)
-        case e: Upstream5xxResponse => (e.reportAs, e.getMessage)
+  private def logError(request: RequestHeader, ex: Throwable): Unit = {
+    def formatPlayExceptionId: Throwable => String = {
+      case p: PlayException => "@" + p.id + " - "
+      case _ => ""
```

Comment:
for 'case _' I'd much rather we return some representation of the Throwable and not ""